clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1123,6 +1124,8 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
       MPM.addPass(SYCLPropagateAspectsUsagePass(/*ExcludeAspects=*/{},
                                                 /*ValidateAspects=*/false));
 
+      MPM.addPass(AddAspectNamesPass());
+
```

Comment:
I think we can say that this is related to metadata cleanup we do, so I would maybe move this closer to `CleanupSYCLMetadataPass`